File: 프로그래머스/unrated/120956. 옹알이 （1）/옹알이 （1）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        String[] patterns = {"aya", "ye", "woo", "ma"};

        for (String word : babbling) {
            String currentWord = word;
            boolean isValid = true;
            String lastPattern = "";

            while (!currentWord.isEmpty() && isValid) {
                boolean foundPattern = false;
                for (String pattern : patterns) {
                    if (currentWord.startsWith(pattern) && !pattern.equals(lastPattern)) {
                        currentWord = currentWord.substring(pattern.length());
                        lastPattern = pattern;
                        foundPattern = true;
                        break;
                    }
                }
                isValid = foundPattern;
            }

            if (isValid && currentWord.isEmpty()) {
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.py
def solution(babbling):
    answer = 0
    oal_list = ["aya", "ye", "woo", "ma"]
    for bab in babbling:
        show_cnt = 0
        for oal in oal_list:
            if oal in bab:
                show_cnt += 1
                bab = bab.replace(oal, "t")
        if show_cnt == len(bab):
            answer += 1
    return answer